### DISCUSSION TITLE
Can i detect if user is new or not in supabase similar to firebase?

### DISCUSSION THREAD

**Discussion by roker15** [supabase/supabase#3245] [Questions]
When user login in firebase, we can determine if user is first time or not. we do something like 
`task.getResult().getAdditionalUserInfo().isNewUser()`
Do we have similar functanility in supabase so that we can perform some initical operation like profile update, assigning roles etc...

**Comment by kiwicopple** [comment]
Hey @roker15 - we don't have that functionality yet, seems like a viable addition to our auth. Thanks for the suggestion

As a workaround, you can implement yourself for now. It would look something like this:

```js
await supabase.auth.update({
  data: {
    new: false
  },
});
```

eg, you can store whichever flag you want in the user data using `update()`: https://supabase.io/docs/reference/javascript/auth-update. If `new == (null || undefined)`, then you would know that this user is "new"

**Reply by vigneshksaithal** [reply]
@kiwicopple is this feature available now?
